### PR TITLE
CLOUDP-266431: Add the output transformation logic to foascli (skeleton)

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-266431

This PR proposes the following changes:

- moves the `changelog.go` from openapi package to its own package
- defines a `outputfilter` package which will contains all the transformations for the entries
- adds the message transformations

```bash
./bin/foascli changelog create -b ~/Downloads/test/base -r ~/Downloads/test/revision -e ~/Downloads/test/exceptions.yaml                                         
Printing the checks
{
  "id": "api-schema-removed",
  "text": "removed the schema 'AzureHardwareSpec20250101'",
  "level": 3,
  "section": "components"
}
{
  "id": "endpoint-added",
  "text": "endpoint added",
  "level": 1,
  "operation": "GET",
  "operationId": "getVersion",
  "path": "/rest/unauth/version",
  "source": "/Users/andrea.angiolillo/Downloads/test/revision/v2.json",
  "section": "paths"
}
....
```

### CODE REVIEW

File: tools/cli/internal/changelog/changelog.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	}
+
+	var entries []*Entry
+	err = json.Unmarshal(bytes, &entries)
+	if err != nil {
+		return nil, err
+	}
+
+	return transformEntries(entries), nil
+}
```
Comment (andreaangiolillo) [inline tools/cli/internal/changelog/changelog.go:123]: similar to oasdiff changelog command, we use the formatter to get the correct output from the checker and then we apply the transformation `transformEntries(entries)`

**Review by blva — APPROVED** [review]

File: tools/cli/internal/changelog/outputfilter/message.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	text = setValueSet(text)
+	text = setValueRemoved(text)
+	text = removeInlineSchemaIndex(text)
+	text = removeRedundantOneOfAllOf(text)
+	entry.Text = text
+
+	return entry
+}
+
+func removeRedundantOneOfAllOf(text string) string {
```
Comment (blva) [inline tools/cli/internal/changelog/outputfilter/message.go:19]: love how all of this is looking much simpler. to help w/ reviews, can you share which of the transformations from the python scripts  you covered here?

Comment (andreaangiolillo) [reply tools/cli/internal/changelog/outputfilter/message.go:19]: this one is from messages.py which I have to say is quite simple also in mms xD

Comment (blva) [reply tools/cli/internal/changelog/outputfilter/message.go:19]: ah OK heheheheh lovely and the message.go makes sense too

**Pull request merged**